packages/components/src/sandbox/index.js

```diff
@@ -215,23 +221,21 @@ export default function Sandbox( {
 		ref.current.addEventListener( 'load', tryNoForceSandbox, false );
 		defaultView.addEventListener( 'message', checkMessageForResize );
 
+		const iframeRef = ref.current;
+
 		return () => {
-			ref.current?.removeEventListener(
-				'load',
-				tryNoForceSandbox,
-				false
-			);
+			iframeRef?.removeEventListener( 'load', tryNoForceSandbox, false );
 			defaultView.addEventListener( 'message', checkMessageForResize );
 		};
-	}, [] );
+	}, [ trySandbox ] );
 
 	useEffect( () => {
 		trySandbox();
-	}, [ title, styles, scripts ] );
+	}, [ trySandbox ] );
 
 	useEffect( () => {
 		trySandbox( true );
-	}, [ html, type ] );
+	}, [ trySandbox ] );
```

Comment:
I'm afraid that the refactor around `trySandbox` doesn't make much sense — it looks like the author's original intent was to pass `true` to the `trySandbox` call only when `html` and `type` changed — with the current changes in this PR, `trySandbox` would be called with both `undefined` and `true` everytime its value changes.